mm2src/coins/eth.rs

```diff
@@ -5053,13 +5055,14 @@ fn get_addr_nonce(addr: Address, web3s: Vec<Web3Instance>) -> Box<dyn Future<Ite
                     return ERR!("Couldn't get nonce after 5 errored attempts, aborting");
                 }
             } else {
-                let max = nonces.iter().max().unwrap();
-                let min = nonces.iter().min().unwrap();
-                if max == min {
-                    return Ok(*max);
-                } else {
-                    warn!("Max nonce {} != {} min nonce", max, min);
-                }
+                let max = nonces.iter().map(|(n, _)| *n).max().unwrap();
```

Comment:
You can use expect here instead of unwrap, writing that nonces is checked above and will not be empty.

Reply:
Done